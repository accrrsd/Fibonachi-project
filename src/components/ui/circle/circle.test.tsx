import { render, screen } from '@testing-library/react'
import { Circle } from './circle'
import { ElementStates } from '../../../types/element-states'

const TestingReactComponent: React.FC<{ dataTest: string }> = ({ dataTest }) => {
  return <div data-testid={dataTest}></div>
}

describe('Circle component', () => {
  it('render circle', () => {
    const { baseElement } = render(<Circle></Circle>)
    expect(baseElement).toBeInTheDocument()
  })

  it('render circle with letter snapshot', () => {
    const circle = render(<Circle letter="a"></Circle>)
    expect(circle).toMatchSnapshot()
  })

  it('render circle with head snapshot', () => {
    const circle = render(<Circle head="head"></Circle>)
    expect(circle.getByText('head')).toBeInTheDocument()
  })

  it('render circle with react element in head', () => {
    render(<Circle head={<TestingReactComponent dataTest="react_element_in_head" />}></Circle>)
    expect(screen.getByTestId('react_element_in_head')).toBeInTheDocument()
  })

  it('render circle with tail', () => {
    const circle = render(<Circle tail="tail"></Circle>)
    expect(circle.getByText('tail')).toBeInTheDocument()
  })

  it('render circle with react element in tail', () => {
    render(<Circle tail={<TestingReactComponent dataTest="react_element_in_tail" />}></Circle>)
    expect(screen.getByTestId('react_element_in_tail')).toBeInTheDocument()
  })

  it('render circle with index', () => {
    const circle = render(<Circle index={123}></Circle>)
    expect(circle.getByText('123')).toBeInTheDocument()
  })

  it('render small circle', () => {
    const { container } = render(<Circle isSmall={true}></Circle>)
    expect(container.querySelector('.small')).not.toBeNull()
  })
  it('render default circle', () => {
    const { container } = render(<Circle state={ElementStates.Default}></Circle>)
    expect(container.querySelector(`.${ElementStates.Default}`)).not.toBeNull()
  })
  it('render changing circle', () => {
    const { container } = render(<Circle state={ElementStates.Changing}></Circle>)
    expect(container.querySelector(`.${ElementStates.Changing}`)).not.toBeNull()
  })
  it('render modified circle', () => {
    const { container } = render(<Circle state={ElementStates.Modified}></Circle>)
    expect(container.querySelector(`.${ElementStates.Modified}`)).not.toBeNull()
  })
})
