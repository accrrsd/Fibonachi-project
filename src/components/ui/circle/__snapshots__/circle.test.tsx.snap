// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle component render circle with letter snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="content "
      >
        <div
          class="text text_type_input text_color_input mb-4 absolute head element"
        />
        <div
          class="circle   default"
        >
          <p
            class="text text_type_circle text_color_input letter"
          >
            a
          </p>
        </div>
        <p
          class="text text_type_input text_color_input mt-4 absolute index"
        />
        <div
          class="text text_type_input text_color_input mt-4 absolute tail30 element"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="content "
    >
      <div
        class="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        class="circle   default"
      >
        <p
          class="text text_type_circle text_color_input letter"
        >
          a
        </p>
      </div>
      <p
        class="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        class="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
